src/Features/ShippingLabelBannerDisplayRules.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * WooCommerce Onboarding Tasks
+ * NOTE: DO NOT edit this file in WooCommerce core, this is generated from woocommerce-admin.
+ *
+ * @package Woocommerce Admin
+ */
+
+namespace Automattic\WooCommerce\Admin\Features;
+
+use \Automattic\WooCommerce\Admin\Loader;
+
+/**
+ * Determines whether or not the Shipping Label Banner should be displayed
+ */
+class ShippingLabelBannerDisplayRules {
+
+	/**
+	 * Minimum supported Jetpack version.
+	 *
+	 * @var string
+	 */
+	private $min_jetpack_version = '4.4';
+
+	/**
+	 * Supported countries by USPS, see: https://webpmt.usps.gov/pmt010.cfm
+	 *
+	 * @var array
+	 */
+	private $supported_countries = array( 'US', 'AS', 'PR', 'VI', 'GU', 'MP', 'UM', 'FM', 'MH' );
+
+	/**
+	 * Array of supported currency codes.
+	 *
+	 * @var array
+	 */
+	private $supported_currencies = array( 'USD' );
+
+	/**
+	 * Determines whether or not the banner should be displayed.
+	 *
+	 * @param string $jetpack_version Installed Jetpack version to check.
+	 */
+	public function should_display_banner( $jetpack_version ) {
+		$woocommerce_shipping_ab_active     = get_option( 'woocommerce_shipping_ab_active' );
+		$woocommerce_product_needs_shipping = false;
+		$woocommerce_store_in_us            = false;
+		$woocommerce_store_currency_in_usd  = false;
+		$base_currency                      = get_woocommerce_currency();
+		$base_location                      = wc_get_base_location();
+
+		if ( $woocommerce_shipping_ab_active ) {
+			return true;
+		}
+
+		if ( $this->banner_dismissed() ) {
+			return false;
+		}
+
+		if ( $this->jetpack_disconnected() ||
+			$this->jetpack_not_up_to_date( $jetpack_version ) ||
+			$this->wcs_not_installed() ||
+			$this->ups_not_installed() ||
+			$this->fedex_not_installed() ||
+			$this->shipstation_not_installed() ||
+			$this->shippingeasy_not_installed() ) {
```

Comment:
With these rules the banner will basically never show.
I think we [actually](https://github.com/Automattic/woocommerce-shipping-issues/issues/13#issuecomment-595928583) want to check if any of these plugins are installed instead of NOT installed.

Reply:
+1
The idea is that users with extensions like Shipstation probably already have an established fulfillment process they use, so there would be less value in us displaying the prompt to them.

Reply:
Ok, thanks guys. I wasn't sure what were the conditions precedence to display the banner. I scanned the related product p2 posts but was difficult to translate that info to the requirements in this issue.

Reply:
As I understand it **ALL** of the following must be true:
* A/B Test Active
* Banner NOT dismissed
* Jetpack >= 4.4 installed and active
* Jetpack Connected
* WCS plugin not installed 
* Shipstation not installed
* UPS not Installed
* Fedex not installed
* Order contains physical products which need shipped
* Store is located in US
* Store currency is set to USD

The only item I'm a little uncertain about is the A/B test. I think the plan was to not use an A/B test at first when released with WooCommerce Admin but when it is eventually integrated into WooCommerce 4.1 we'd use a 50/50 or other A/B test. I don't remember where I saw that but I think for now it means we don't need to worry about an A/B test.

Reply:
@superdav42,

In regards to the A/B tests, we are planning on targeting 25% of relevant sites when we initially launch: https://github.com/Automattic/woocommerce-shipping-issues/issues/11

We are rushing to get the core functionality of the prompt complete, so I have this marked as a lower priority, but so far this is still the plan

Reply:
[I have a A/B test branch based on this branch](https://github.com/woocommerce/woocommerce-admin/compare/shipping-banner...add/ab_test?expand=1) I'll make a PR after this merges in case any big changes are made. You can disregard the A/B check in this PR since I'll add it in that branch.

So we actually want to display the banner on sites with WCS installed that have **not** accepted the terms of service. The banner will be a friendlier way for merchant to accept the ToS and launch the shipping modal.

So (excluding the A/B test which I'll handle on the other branch), the conditions should be:

- Banner NOT dismissed
- Jetpack >= 4.4 installed and active
- Jetpack Connected
- No incompatible extensions installed, so
   - Shipstation not installed
   - UPS not Installed
   - Fedex not installed
   - ShippingEasy not installed
- Order contains physical products which need to be shipped ([we should check that the order status is not set to complete](https://a8c.slack.com/archives/CKNC7KACA/p1584395171186200))
- Store is located in US
- Store currency is set to USD
- `WCS plugin not installed` OR` WCS is installed *AND* ToS have NOT been accepted *AND* WCS version is 1.22.5 or greater`
    - The 1.22.5 or greater requirement is so we can launch the shipping modal from the banner

Reply:
Note: we should be able to check the ToS from wc-admin by calling WCS's `WC_Connect_Options::get_option( 'tos_accepted' )`